src/service/index.ts

```diff
@@ -31,6 +32,10 @@ const processMovieResult = async (movies: MovieResult[], overseerr: OverseerrApi
 
                     await overseerr.requestMovie(movie.id, options);
                 }
+                numAdded += 1;
+                if (maxRequests !== undefined && numAdded >= maxRequests) {
+                    break;
```

Comment:
I'd recommend adding a log statement
```suggestion
                    logger.info(`Maximum requests limit (${maxRequests}) reached. Stopping analyze here...`);
                    break;
```